### PR TITLE
Feature/add processor dynamically

### PR DESCRIPTION
**Pull request by sindelio**
Finally ready to receive reviews!
Guys, please rebuild the whole system after you pull this code. 
Remembering that now the processor Quorum node is separated from the controller Quorum node after the setup. If you  want to have them on the same Quorum network, them use the endpoint in the management to add a processor. In order to do that, the account address and the enode of the processor are required. The account address of the processor can be found in the file ```/quorum/generated_configs/node2/dd/account.txt``` and the enode in the file ```quorum/generated_configs/node2/static_nodes_entry```.
It's a good idea for me to give another quick demo on this Tuesday.
Have a good weekend and holiday o/

### CODE REVIEW

File: cg/src/domains/processors/processors.service.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const { db } = require('../../db');
+
+class ProcessorsService {
+  constructor(database = db) {
+    this.db = database;
+  }
```
Comment (sindelio) [inline cg/src/domains/processors/processors.service.js]: This is empty for now, we may change it in the future if we decide to automate the entire flow of adding a processor to the network

File: cg/src/utils/blockchain.js [hunk 2]
```diff
@@ -188,12 +199,10 @@ async function listenerForConsentEvent(callback) {
       winston.error(`Error handling consent given to ${error.toString()}`);
       return;
     }
-    if (
+    if (data.returnValues.processorsConsented) {
```
Comment (sindelio) [inline cg/src/utils/blockchain.js]: .map was being invoked in a situation where the processorsConsented could be null, this is a simple fix for that

File: docker-compose.yml [hunk 3]
```diff
@@ -108,7 +108,7 @@ services:
       quorum_net:
         ipv4_address: '172.13.0.2'
   constellation2:
-    image: cleargdpr/quorum:0.0.1
+    image: cleargdpr/quorum_processor:0.0.2
```
Comment (sindelio) [inline docker-compose.yml]: new image for the processors

File: quorum/scripts/generate_env_vars.sh [hunk 4]
```diff
@@ -14,7 +14,8 @@ if [ "$1" = "" ] ; then
 fi
 
 GENESIS_JSON=`node "$pwd/quorum/scripts/generate_genesis.js" $@`
-STATIC_NODES=`node "$pwd/quorum/scripts/generate_static_nodes.js" $@`
+STATIC_NODES=`node "$pwd/quorum/scripts/generate_static_nodes.js" $1`
```
Comment (sindelio) [inline quorum/scripts/generate_env_vars.sh]: Mocking the separation between the controller and the processor by making them have differente static-nodes files

File: quorum/start-node.sh [hunk 5]
```diff
@@ -69,7 +69,7 @@ elif [ "$SERVICE" = "geth" ]; then
     /usr/local/bin/geth --datadir /qdata/dd init /qdata/genesis.json
   fi
 
-  GETH_ARGS="--datadir /qdata/dd --raft --rpc --rpcaddr 0.0.0.0 --port "$ETH_PORT" --ws --wsorigins="*" --wsaddr 0.0.0.0 --wsport "$WEBSOCKET_PORT" --rpcport "$RPC_PORT" --raftport "$RAFT_PORT" --wsapi admin,db,eth,debug,miner,net,shh,txpool,personal,web3,quorum --rpcapi admin,db,eth,debug,miner,net,shh,txpool,personal,web3,quorum --nodiscover --unlock 0 --password /qdata/passwords.txt"
+  GETH_ARGS="--datadir /qdata/dd --raft --raftjoinexisting 2 --rpc --rpcaddr 0.0.0.0 --port "$ETH_PORT" --ws --wsorigins="*" --wsaddr 0.0.0.0 --wsport "$WEBSOCKET_PORT" --rpcport "$RPC_PORT" --raftport "$RAFT_PORT" --wsapi admin,db,eth,debug,miner,net,shh,txpool,personal,web3,raft --rpcapi admin,db,eth,debug,miner,net,shh,txpool,personal,web3,raft --nodiscover --unlock 0 --password /qdata/passwords.txt"
```
Comment (sindelio) [inline quorum/start-node.sh]: This is the command that starts Geth in the processor. This file goes inside the processors Docker image

File: setup_processor.js [hunk 6]
```diff
@@ -0,0 +1 @@
+console.log(process.env);
```
Comment (sindelio) [inline setup_processor.js]: Working on this

File: api/src/routes/products.js [hunk 7]
```diff
@@ -1,22 +0,0 @@
-const express = require('express');
```
Comment (sindelio) [inline api/src/routes/products.js]: Removing this unused file

File: cg/src/domains/management/processors/processors.validators.js [hunk 8]
```diff
@@ -8,7 +8,10 @@ const addProcessorValidator = celebrate({
     logoUrl: Joi.string().max(255),
     description: Joi.string(),
     scopes: Joi.array().items(Joi.string()),
-    address: Joi.string().regex(/^0x[\da-fA-F]{40}$/)
+    enode: Joi.string()
```
Comment (sindelio) [inline cg/src/domains/management/processors/processors.validators.js]: Consider putting a regexp for the enode later

File: cg/src/utils/blockchain.js [hunk 9]
```diff
@@ -69,7 +86,7 @@ async function getIsErased(subjectId) {
 async function getSubjectDataState(subjectId, processor = controllerAddress) {
   const quorumContract = await getContract();
   let result = await quorumContract.methods.getSubjectDataState(subjectId, processor).call();
-  return +result; //'+' coerces the returned value to a string
+  return +result; //'+' coerces the returned value to a number?
```
Comment (sindelio) [inline cg/src/utils/blockchain.js]: Coerces to a number right?

Comment (Nelrohd) [reply cg/src/utils/blockchain.js]: or `Number(result)` | `parseInt(result, 10)`?

**Pull request merged**